### DISCUSSION TITLE
Camel Case with single quotes doesnt work as expected

### DISCUSSION THREAD

**Discussion by bobrowadam** [minad/jinx#177] [Q&A] [answered]
Hi,
When working in a typescript-mode buffer, jinx doesn't identify camel case strings wrapped with single quotes. double quoting works fine.
'camelCase' => marked as incorrect
"camelCase"  => correct

I tried to debug and I think that the problem is in this line:
https://github.com/minad/jinx/blob/09969c50b965a9a01f3c02ec47c4496b991a8624/jinx.el#L465
The regex "\\([[:upper:]]?[[:lower:]]+\\)\\(?:[[:upper:]][[:lower:]]+\\)+\\>" doesn't match when the word wrapped with single quotes.

Thanks

**Comment by minad** [comment, marked as answer]
I can reproduce this problem. I think the problem is that Jinx has special handling for apostrophes at the beginning or end of words. See this code right before the regexp, you've mentioned:

https://github.com/minad/jinx/blob/09969c50b965a9a01f3c02ec47c4496b991a8624/jinx.el#L452-L460

I am happy to take a patch, if we figure out a way to handle this edge case. Otherwise I consider this a low priority issue, given that I hope to preserve the existing apostrophe handling.

EDIT: I gave it a try and the problem is exactly as you described. It is simple: The regexp doesn't match since the single quote is treated as part of words (apostrophe). I am afraid we cannot easily fix this by default. However it should be possible to reconfigure the single quote as non-word character in the `jinx--syntax-table`, if you accept the downside that quotes are then not considered parts of words anymore, which may lead to false positives.

**Reply by bobrowadam** [reply]
Thanks for the reply!
As you mentioned, I'll try to configure the syntax-table and see how bad the experience is with word apostrophes.
If I come out with a fix I'll be happy to send a PR.

**Comment by fireirohuncle** [comment]
Is there any solution to enable camelCase checking for a particular font between single quotes? For example, in prog-mode, I rarely expect to see apostrophes in font-lock-string-face, but I encounter camelCase quite often.
Or perhaps there is some another workaround?
